### PR TITLE
Linting

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
Well, this mushroomed.

Originally a PR to add linting it grew into adding additional config packages to get linting to work and then I had to do that thing where you have to solve all the linting errors causing a cascade of changes.

### What linting errors were there?
Most changes were classic things like single quotes vs. double or spacing but the recommended configuration for @guardian repos also has things like no default exports which caused errors here too

I also updated the `scripts` section of `package.json`, giving:

```json
    "lint": "eslint . --ext .ts,.tsx",
    "test": "jest",
    "tsc": "tsc",
    "validate": "npm run lint && npm run test && npm run tsc",
```

### CODE REVIEW

**Review by shtukas — APPROVED** [review]

File: src/index.ts [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
+export type { Image } from "./image";
+export { Role } from "./image";
+export type { Lightbox } from "./lightbox";
+export type { Sizes } from "./sizes";
 
 // ----- Components ----- //
 
-export { default as BodyImage } from './components/bodyImage';
-export { default as FigCaption } from './components/figCaption';
-export { default as Img } from './components/img';
```
Comment (mxdvl) [inline src/index.ts:13]: Is this going to work?

**Review by mxdvl — COMMENTED** [review]
Do you need both `.eslintrc` and `.prettierrc`?

File: src/index.ts (hunk 1 above)
Comment (oliverlloyd) [reply src/index.ts:13]: In the same file further up, we're already doing `export { Role } from './image';` so `export { Img } from './components/img';` should work the same.

**Comment by oliverlloyd** [comment]
> Do you need both `.eslintrc` and `.prettierrc`?
> 

To be honest, I've never really got my head around how and where eslint and prettier overlap. I'm used to having the two different config files though and the `onfigs` repo seems to take the same approach so I'm happy to defer to the standard path

> Should we also add the `.editorconfig`? https://github.com/guardian/configs/blob/main/.editorconfig
> 
I think this makes sense. We could perhaps bring this in in a different PR to try to limit the noice from this one

**Comment by mxdvl** [comment]
Glad I'm not the only one struggling to grasp the distinction between Prettier and ESLint 😆

I think it should be set up so that prettier runs after ESLint, possibly trigger by it.

**Review by gtrufitt — APPROVED** [review]
🆒

**Comment by oliverlloyd** [comment]
[Linting for IR](https://trello.com/c/9r2nDVtM/2170-linting-for-ir)

**Pull request merged**